port/unix/omrsysinfo.c

```diff
@@ -3568,18 +3585,31 @@ omrsysinfo_startup(struct OMRPortLibrary *portLibrary)
 	 */
 	if (0 == attachedPortLibraries) {
 		if (omrthread_monitor_init_with_name(&cgroupEntryListMonitor, 0, "cgroup entry list monitor")) {
-			return OMRPORT_ERROR_STARTUP_SYSINFO_MONITOR_INIT;
+			rc = OMRPORT_ERROR_STARTUP_SYSINFO_MONITOR_INIT;
+			goto _end;
 		}
 	}
 	attachedPortLibraries += 1;
-	isRunningInContainer(portLibrary, &PPG_isRunningInContainer);
-#endif /* defined(LINUX) */
-	return 0;
 
-#if (defined(S390) || defined(J9ZOS390))
-	PPG_stfleCache.lastDoubleWord = -1;
-#endif
+	if (isCgroupV1Available(portLibrary)) {
+		PPG_sysinfoControlFlags |= OMRPORT_SYSINFO_CGROUP_V1_AVAILABLE;
+	} else if (isCgroupV2Available()) {
+		PPG_sysinfoControlFlags |= OMRPORT_SYSINFO_CGROUP_V2_AVAILABLE;
+	}
+	/* isCgroupV1Available must be called before the following function as the latter checks
+	 * a bit set by the former.
+	 */
+	if (0 == isRunningInContainer(portLibrary, &runningInContainer)) {
+		if (runningInContainer) {
+			PPG_sysinfoControlFlags |= OMRPORT_SYSINFO_RUNNING_IN_CONTAINER;
+		}
+	} else {
+		rc = OMRPORT_ERROR_STARTUP_SYSINFO_RUNNING_IN_CONTAINER;
+	}
 
+_end:
+#endif /* defined(LINUX) */
```

Comment:
This should match the `#if` statement.
```suggestion
#endif /* defined(LINUX) && !defined(OMRZTPF) */
```